packages/superset-ui-chart/src/components/createLoadableRenderer.ts

```diff
@@ -0,0 +1,52 @@
+import Loadable from 'react-loadable';
+
+export type LoadableRendererProps = {
+  onRenderFailure?: Function;
+  onRenderSuccess?: Function;
+};
+
+const defaultProps = {
+  onRenderFailure() {},
+  onRenderSuccess() {},
+};
+
+export interface LoadableRenderer<Props, Exports>
+  extends React.ComponentClass<Props & LoadableRendererProps>,
+    Loadable.LoadableComponent {}
+
+export default function createLoadableRenderer<Props, Exports>(
+  options: Loadable.OptionsWithMap<Props, Exports>,
+): LoadableRenderer<Props, Exports> {
+  // eslint-disable-next-line babel/new-cap
+  const LoadableRenderer = Loadable.Map(options) as LoadableRenderer<Props, Exports>;
+
+  // Extends the behavior of LoadableComponent to provide post-render listeners
+  class CustomLoadableRenderer extends LoadableRenderer {
+    static defaultProps: object;
+
+    componentDidMount() {
+      this.afterRender();
+    }
+
+    componentDidUpdate() {
+      this.afterRender();
+    }
+
+    afterRender() {
+      const { loaded, loading, error } = this.state;
+      const { onRenderFailure, onRenderSuccess } = this.props;
+      if (!loading) {
+        if (error) {
+          (onRenderFailure as Function)(error);
```

Comment:
casting here because it doesn't catch that they will be defined from `defaultProps`. 

I initially had a check that they were defined, but then branch coverage will never be 100% because they'll always be there (unless we force an invalid prop value of `null`, which we could just `@ts-ignore`)